src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseFVMKernels.hpp

```diff
@@ -47,8 +47,582 @@ using namespace constitutive;
+                                                  kernelComponent.numPointsInFlux( iconn ) );
+
+      kernelComponent.setup( iconn, stack );
+      kernelComponent.computeFlux( iconn, stack );
+      kernelComponent.complete( iconn, stack );
+    } );
+  }
+
+
+protected:
+
+  // Stencil information
+
+  /// Reference to the stencil wrapper
+  STENCILWRAPPER const m_stencilWrapper;
+
+  /// Connection to element maps
+  typename STENCILWRAPPER::IndexContainerViewConstType const m_seri;
+  typename STENCILWRAPPER::IndexContainerViewConstType const m_sesri;
+  typename STENCILWRAPPER::IndexContainerViewConstType const m_sei;
+};
+
+/**
+ * @class FaceBasedAssemblyKernelFactory
+ */
+class FaceBasedAssemblyKernelFactory
+{
+public:
+
+  /**
+   * @brief Create a new kernel and launch
+   * @tparam POLICY the policy used in the RAJA kernel
+   * @tparam STENCILWRAPPER the type of the stencil wrapper
+   * @param[in] rankOffset the offset of my MPI rank
+   * @param[in] dofKey string to get the element degrees of freedom numbers
+   * @param[in] solverName name of the solver (to name accessors)
+   * @param[in] elemManager reference to the element region manager
+   * @param[in] stencilWrapper reference to the stencil wrapper
+   * @param[in] dt time step size
+   * @param[inout] localMatrix the local CRS matrix
+   * @param[inout] localRhs the local right-hand side vector
+   */
+  template< typename POLICY, typename STENCILWRAPPER >
+  static void
+  createAndLaunch( globalIndex const rankOffset,
+                   string const & dofKey,
+                   string const & solverName,
+                   ElementRegionManager const & elemManager,
+                   STENCILWRAPPER const & stencilWrapper,
+                   real64 const & dt,
+                   CRSMatrixView< real64, globalIndex const > const & localMatrix,
+                   arrayView1d< real64 > const & localRhs )
+  {
+    integer constexpr NUM_DOF = 1;
+
+    ElementRegionManager::ElementViewAccessor< arrayView1d< globalIndex const > > dofNumberAccessor =
+      elemManager.constructArrayViewAccessor< globalIndex, 1 >( dofKey );
+    dofNumberAccessor.setName( solverName + "/accessors/" + dofKey );
+
+    using kernelType = FaceBasedAssemblyKernel< NUM_DOF, STENCILWRAPPER >;
+    typename kernelType::SinglePhaseFlowAccessors flowAccessors( elemManager, solverName );
+    typename kernelType::SinglePhaseFluidAccessors fluidAccessors( elemManager, solverName );
+    typename kernelType::PermeabilityAccessors permAccessors( elemManager, solverName );
+
+    kernelType kernel( rankOffset, stencilWrapper, dofNumberAccessor,
+                       flowAccessors, fluidAccessors, permAccessors,
+                       dt, localMatrix, localRhs );
+    kernelType::template launch< POLICY >( stencilWrapper.size(), kernel );
+  }
+};
+
 /******************************** FluxKernel ********************************/
 
+// To delete it after verifying FaceBasedAssemblyKernel
```

Comment:
Can delete now?

Reply:
> Can delete now?

Not yet, since this old kernels and its child struct(s) are still being used. It will need extra work to replace and remove them. I can do it now or leave to future.

Reply:
No need to do it now, I was just making sure we're not keeping unused code around